File: ios/StillMoment/Domain/Models/ResolvedAttunement.swift
//
//  ResolvedAttunement.swift
//  Still Moment
//
//  Domain - Resolved attunement audio (built-in or custom)
//

import Foundation

/// A resolved attunement audio entry, regardless of source (built-in or user-imported).
///
/// Consumers use this instead of checking Introduction + CustomAudioRepository separately.
/// The resolver transparently handles both sources.
struct ResolvedAttunement: Equatable {
    /// Unique identifier (Introduction.id for built-in, UUID string for custom)
    let id: String

    /// Localized display name
    let displayName: String

    /// Audio duration in seconds
    let durationSeconds: Int
}

File: android/app/src/main/kotlin/com/stillmoment/domain/models/ResolvedAttunement.kt
package com.stillmoment.domain.models

/**
 * A resolved attunement audio entry, regardless of source (built-in or user-imported).
 *
 * Consumers use this instead of checking Attunement + CustomAudioRepository separately.
 * The resolver transparently handles both sources.
 *
 * @property id Unique identifier (Attunement.id for built-in, UUID string for custom)
 * @property displayName Localized display name
 * @property durationSeconds Audio duration in seconds
 */
data class ResolvedAttunement(
    val id: String,
    val displayName: String,
    val durationSeconds: Int
)
